### DISCUSSION TITLE
New template format

### DISCUSSION THREAD

**Discussion by frapentiny** [mampfes/hacs_waste_collection_schedule#890] [Q&A]
Hi,

I'm new to home assistant and found a hint in the documentation, that the legacy format for sensor configuration still works but is no longer recommended.
Is there a way to use the modern sensor configuration format with waste_collection_schedule?

_- platform: waste_collection_schedule
    name: AbfallBiotonnenext
    details_format: "upcoming"
    value_template: 'am: {{value.date.strftime("%d.%m.%Y")}}'
    types:
      - BioTonne_

I fail with the transformation of the above legacy format, because "platform" does not exits in the modern format.

How should the modern sensor format look like?

Thanks in advance for your support.

Cheers
Frank

**Comment by mampfes** [comment]
Could you add a link to the documentation you are refering to?

Just follow the waste_collection_schedule documentation regarding configuration and sensor setup. There are a lot of examples in the generic part which describes - amongst others - the sensor configuration and also in the source specific documentation.

**Reply by frapentiny** [reply]
Here you can find the note that the old format is no longer recommended:
https://www.home-assistant.io/integrations/template/#legacy-binary-sensor-configuration-format

I looked at the examples and got them to work, but now I wanted to bring them into the modern format, which I was not able to do.

**Comment by mampfes** [comment]
This documentation applies only to the `template` integration and is not applicable for other integrations (like waste_collection_schedule).

**Reply by frapentiny** [reply]
thanks for the hint. I will use the examples that already work.

**Reply by Bhoft** [reply]
So you are saying it's not possible?

I was struggling with the same issue because of that text
https://www.home-assistant.io/integrations/template/#legacy-sensor-configuration-format

so you are saying that the sensor code from 
https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/configuration.yaml
can't be changed into the new format?

**Reply by mampfes** [reply]
Correct
